cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java

```diff
@@ -92,42 +95,69 @@ private static void setupMetrics() throws Exception {
+    verifySearchResult(getContextSearchUri(YOUR_SPACE, "WCount1"), ImmutableList.<String>of("b", "f", "p"));
+    verifySearchResult(getContextSearchUri(MY_SPACE, "WCount1"), ImmutableList.<String>of());
+
+    // verify other metrics for WCount app
+    verifySearchResult(getContextSearchUri(YOUR_SPACE, "WCount1.b.ClassicWordCount"),
                        ImmutableList.<String>of("m", "r"));
-    verifySearchResult("/v3/metrics/search?target=childContext&context=WCount1.b.ClassicWordCount.m",
-                       ImmutableList.<String>of());
+    verifySearchResult(getContextSearchUri(YOUR_SPACE, "WCount1.b.ClassicWordCount.m"), ImmutableList.<String>of());
   }
 
   @Test
   public void testQueryMetrics() throws Exception {
-    // aggregate result
+    // aggregate result, in the right namespace
     verifyAggregateQueryResult(
-      "/v3/metrics/query?context=WCount1.f.WCounter.splitter&metric=system.reads&aggregate=true", 3);
+      getMetricsAggregateQueryUri(YOUR_SPACE, "WCount1.f.WCounter.splitter", "system.reads"), 3);
     verifyAggregateQueryResult(
-      "/v3/metrics/query?context=WCount1.f.WCounter.counter&metric=system.reads&aggregate=true", 1);
+      getMetricsAggregateQueryUri(YOUR_SPACE, "WCount1.f.WCounter.counter", "system.reads"), 1);
+
+    // aggregate result, in the wrong namespace
+    verifyAggregateQueryResult(getMetricsAggregateQueryUri(MY_SPACE, "WCount1.f.WCounter.splitter", "system.reads"), 0);
 
     // time range
     // now-60s, now+60s
-    verifyRangeQueryResult(
-      "/v3/metrics/query?context=WCount1.f.WCounter.splitter&metric=system.reads&start=now%2D60s&end=now%2B60s", 2, 3);
+    verifyRangeQueryResult(getMetricsRangeQueryUri(YOUR_SPACE, "WCount1.f.WCounter.splitter", "system.reads",
+                                                   "now%2D60s", "now%2B60s"), 2, 3);
     // note: times are in seconds, hence "divide by 1000";
-    long start = (emitTs - 60 * 1000) / 1000;
-    long end = (emitTs + 60 * 1000) / 1000;
-    verifyRangeQueryResult(
-      "/v3/metrics/query?context=WCount1.f.WCounter.splitter&metric=system.reads&start=" + start + "&end=" + end, 2, 3);
+    String start = String.valueOf((emitTs - 60 * 1000) / 1000);
+    String end = String.valueOf((emitTs + 60 * 1000) / 1000);
+    verifyRangeQueryResult(getMetricsRangeQueryUri(YOUR_SPACE, "WCount1.f.WCounter.splitter", "system.reads",
+                                                   start, end), 2, 3);
+    // range query, in the wrong namespace
+    verifyRangeQueryResult(getMetricsRangeQueryUri(MY_SPACE, "WCount1.f.WCounter.splitter", "system.reads",
+                                                   start, end), 0, 0);
   }
 
   @Test
   public void testSearchMetrics() throws Exception {
-    verifySearchResult("/v3/metrics/search?target=metric&context=WordCount1.f.WordCounter.splitter",
+    // metrics in myspace
+    verifySearchResult(getMetricsSearchUri(MY_SPACE, "WordCount1.f.WordCounter.splitter"),
```

Comment:
wanted to call out the difference between the two successive calls. If it is within the URI, it may be hard to distinguish why one should fail and the other should succeed.